clippy_lints/src/from_instead_of_into.rs

```diff
@@ -0,0 +1,119 @@
+                let target_bound = &bounds[position];
+                if let Some(tr_ref) = target_bound.trait_ref();
+                if let Some(def_id) = tr_ref.trait_def_id();
+                if let Some(last_seg) = tr_ref.path.segments.last();
+                if let Some(generic_arg) = last_seg.args().args.first();
+                if let Some(bounded_ty) = snippet_opt(cx, wbp.bounded_ty.span);
+                if let Some(generic_arg_of_from_or_try_from) = snippet_opt(cx, generic_arg.span());
+                then {
+                    let replace_trait_name;
+                    let target_trait_name;
+                    if cx.tcx.is_diagnostic_item(sym::from_trait, def_id) {
+                        replace_trait_name = "Into";
+                        target_trait_name = "From";
+                    } else if cx.tcx.is_diagnostic_item(sym::try_from_trait, def_id) {
+                        replace_trait_name = "TryInto";
+                        target_trait_name = "TryFrom";
+                    } else {
+                        return;
+                    }
+                    let message = format!("{} trait is preferable than {} as a generic bound", replace_trait_name, target_trait_name);
+                    let switched_predicate = format!("{}: {}<{}>", generic_arg_of_from_or_try_from, replace_trait_name, bounded_ty);
+
+                    let low;
+                    if position == 0 {
+                        low = wp.span().lo();
+                    } else {
+                        let previous_bound = &bounds[position -1];
+                        low = previous_bound.span().hi();
+                    }
+                    let removed_span = target_bound.span().with_lo(low);
+
+                    span_lint_and_then(
+                        cx,
+                        FROM_INSTEAD_OF_INTO,
+                        wp.span(),
+                        &message,
+                        |diag| {
+                            diag.span_suggestion(
+                                removed_span,
+                                &format!("remove {} bound", target_trait_name),
+                                "".to_string(),
+                                Applicability::MaybeIncorrect,
+                            );
+
+                            let sugg;
+                            if bounds.len() == 1 {
+                                sugg = switched_predicate;
+                            } else {
+                                sugg = format!(", {}", switched_predicate);
+                            }
+                            diag.span_suggestion(
+                                wp.span().with_lo(wp.span().hi()),
+                                "Add this bound predicate",
+                                sugg,
+                                Applicability::MaybeIncorrect,
+                            );
+                        }
+                    );
```

Comment:
When I handle suggestion with this two `span_suggestion` call the diagnostic output is strange.  Remove suggestion is weird because nothing show up in the span only a `,` left. replacing suggestion is weird because the old span is used. Can you take a look? @flip1995 .

Reply:
Could you tell me why you suggest using multi suggestions? Do you intent to suggest using `multispan_sugg`? If yes, why do you choose it?